### PR TITLE
Avoid boost operators.

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge or @vekterli PR

### CODE REVIEW

**Review by toregge — APPROVED** [review]

File: storage/src/vespa/storageframework/generic/clock/time.h [hunk 1]
```diff
@@ -45,37 +44,34 @@ uint64_t getRawMicroTime(const Clock&);
  * automatic conversion between time types.
  */
 template<typename Type, int MicrosPerUnit>
-class Time : public boost::operators<Type>
+class Time
```
Comment (vekterli) [inline storage/src/vespa/storageframework/generic/clock/time.h:47]: Looks like the majority of member functions here could safely be tagged as `noexcept`

File: storage/src/vespa/storageframework/generic/clock/time.h [hunk 2]
```diff
@@ -170,4 +166,39 @@ inline MicroSecTime MilliSecTime::getMicros() const {
...
+    result -= b;
+    return result;
+}
+
+inline SecondTime
+operator - (SecondTime a, SecondTime b) {
+    SecondTime result(a);
+    result -= b;
+    return result;
+}
```
Comment (vekterli) [inline storage/src/vespa/storageframework/generic/clock/time.h:202]: These could probably be generalized as a single overload per operator, using the "micros per unit" template arg to properly normalize values. But that's basically what `std::chrono` already does, so no point if we're already replacing with that.

Comment (baldersheim) [reply storage/src/vespa/storageframework/generic/clock/time.h:202]: That happens further down the PR pipeline.

File: storage/src/vespa/storageframework/generic/clock/time.h (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storageframework/generic/clock/time.h:47]: That happens further down the PR pipeline.

**Pull request merged**